services/py-api/src/database/transaction_manager.py

```diff
@@ -124,24 +128,33 @@ async def callback(session):
         https://www.youtube.com/watch?v=q6ujWWaRdbA
         """
 
+        """
+        If you want to read more of how the start_session() method works, and how the sessions are
+        managed by the client and the server consider this read:
+        https://github.com/mongodb/specifications/blob/master/source/sessions/driver-sessions.md
+        """
         session = await self._client.start_session()
+        session_id = session.session_id["id"].hex()
+
         try:
             session.start_transaction()
-            LOG.debug("Starting transaction")
+            LOG.debug("Starting transaction", session_id=session_id)
 
             result = await self._retry_tx(callback, *args, session=session, **kwargs)
             if is_err(result):
-                LOG.warning("Aborting transaction due to err {}".format(result.err_value))
+                LOG.warning(f"Aborting transaction due to err {result.err_value}", session_id=session_id)
                 await session.abort_transaction()
                 return result
 
             await self._retry_commit(session)
-            LOG.debug("Transaction commited")
+            LOG.debug("Transaction commited", session_id=session_id)
 
             return result
 
         except Exception as e:
-            LOG.exception("Aborting transaction due to err {}".format(e))
+            # FIXME: When there is a repeating team name or email the below message prints "Aborting transaction with id: {session_id} due to err {team name/email}"
+            # instead of printing a meaningful error
+            LOG.exception(f"Aborting transaction due to err {e}", session_id=session_id)
```

Comment:
If there is an expected error, it is with a warning; we have it above. Remove the FIXME. We log the exception as this is an unexpected err.

Reply:
alright

Reply:
Fixed in 62ecace76fd209bffa772b0238adedf16bd38dc2